src/server/game/World/World.cpp

```diff
@@ -3293,6 +3289,14 @@ CharacterInfo const* World::GetCharacterInfo(ObjectGuid const& guid) const
     return nullptr;
 }
 
+ObjectGuid World::GetCharacterGuidByName(std::string const& name) const
```

Comment:
ObjectAccessor is used for accessing ingame units (players and other entities). ObjectMgr is used to manage / store world database. The purpose of this container is to store name -> guid for all players (even those offline) and I assume this the initial reason why CharacterInfo was placed in World (which also bad place for this). The best solution would be to create a separate manager to handle the cached db data.